test/fixtures/runtime-messaging-extension/background.js

```diff
@@ -33,6 +33,12 @@ browser.runtime.onMessage.addListener((msg, sender, sendResponse) => {
       // a reply from the second listener.
       return false;
 
+    case "test - sendMessage with returned rejected Promise with Error value":
+      return Promise.reject(new Error("rejected-error-value"));
+
+    case "test - sendMessage with returned rejected Promise with non-Error value":
+      return Promise.reject("rejected-non-error-value");
```

Comment:
Try `Promise.reject({message: "rejected-non-error-value"})` instead.

If you wish, keep the existing test case, but check that no useful error message is passed through.

Reply:
If keeping the existing test case, maybe check specifically for an Error with the message "An unexpected error occurred"?

Reply:
For what it's worth, I think this mistake was already present in the code, not introduced by the pull request.

Reply:
I added both the cases, to be fair I think that the behavior that Firefox have when an extension code rejects a plain object with a message property is actually an involuntary side-effect of the behavior we wanted when an API implementation rejects that kind of plain objects: 

https://searchfox.org/mozilla-central/search?q=Promise.reject(%7Bmessage%3A&case=false&regexp=false&path= 

Anyway, the polyfill is currently matching this behavior and I added an explicit test case for it.

@james-cnz Like I described in one of the above comments I'm not asserting the "An unexpected error occurred" error message because Firefox 59 and Firefox 60 have different behaviors (and so the test was failing on travis only on Firefox).